api.go

```diff
@@ -228,9 +228,15 @@ func RecoverCluster(conf *Config, fsm FSM, logs LogStore, stable StableStore,
 		return err
 	}
 
-	// Sanity check the Raft peer configuration.
-	if err := checkConfiguration(configuration); err != nil {
-		return err
+	// Sanity check the Raft peer configuration. Note that it's ok to
+	// recover to an empty configuration if you want to keep the data
+	// but not allow a given server to participate any more (you'd have
+	// to recover again, or add it back into the existing quorum from
+	// another server to use this server).
```

Comment:
Consul needed this - when you gracefully leave a cluster it used to blank out the peers. This isn't strictly necessary to do, but until we take the "go away" change later to this this to stop trying to elect, this server will be a nuisance in the Raft logs. This also seemed like a reasonable way to button up a server that was explicitly removed from a cluster with a final snapshot. I'd be open to removing this eventually, but it's good for Consul in the meantime, and seemed like a low impact on the interface.